edge/pkg/devicetwin/dtmanager/membership.go

```diff
@@ -283,22 +279,18 @@ func removeDevice(context *dtcontext.DTContext, toRemove []dttype.Device, baseMe
 		}
 		//todo
 		context.DeviceList.Delete(device.ID)
-		context.DeviceMutex.Delete(device.ID)
 		if delta {
 			context.Unlock(device.ID)
 		}
+		context.DeviceMutex.Delete(device.ID)
```

Comment:
If the device is removed from the `DeviceMutex` first, then it may result in the context not being unlocked.
https://github.com/kubeedge/kubeedge/blob/b512232fd7f9f576540b912a3b22e51c0aa51f80/edge/pkg/devicetwin/dtcontext/dtcontext.go#L98-L107